### DISCUSSION TITLE
AutoGenView and RouteView issue

### DISCUSSION THREAD

**Discussion by SamEsmailR** [apache/cloudstack#9089] [Q&A] [answered]
i am constantly getting these errors when i am trying to start the development server for the ui 

 ERROR  Failed to compile with 1 error                                                                         8:29:13 PM

 error  in ./src/layouts/RouteView.vue?vue&type=script&lang=js

Syntax Error: TypeError: Cannot read properties of null (reading 'content')

 @ ./src/layouts/RouteView.vue?vue&type=script&lang=js 1:0-280 1:0-280 1:281-550 1:281-550
 @ ./src/layouts/RouteView.vue
 @ ./src/layouts/index.js
 @ ./src/config/router.js
 @ ./src/router/index.js
 @ ./src/main.js
 @ multi (webpack)-dev-server/client?http://192.168.1.5:5050&sockPath=/sockjs-node (webpack)/hot/dev-server.js ./src/main.js                 

 ERROR  Failed to compile with 1 error                                                                         8:28:49 PM

 error  in ./src/views/AutogenView.vue?vue&type=script&lang=js

Syntax Error: TypeError: Cannot read properties of null (reading 'content')

 @ ./src/views/AutogenView.vue?vue&type=script&lang=js 1:0-282 1:0-282 1:283-554 1:283-554
 @ ./src/views/AutogenView.vue
 @ ./src/config/router.js
 @ ./src/router/index.js
 @ ./src/main.js
 @ multi (webpack)-dev-server/client?http://192.168.1.5:5050&sockPath=/sockjs-node (webpack)/hot/dev-server.js ./src/main.js

i am using windows 10 with nodejs 16 installed this is the .env.local CS_URL=http://localhost:8080
and thats it i made no other changes

**Comment by DaanHoogland** [comment]
@SamEsmailR, not sure if this helps:
```
$ npm -version
14:53:26
8.19.4
```

**Comment by yadvr** [comment]
Hi @SamEsmailR I use nodejs v16 on Mac OSX and Ubuntu 22.04/24.04, and I don't face any issues. Was this the first time you've tried this or do you have any local changes you were trying to test/build. If you're not able to fix, as Windows isn't the usual development env most of us have, you can try to setup a Ubuntu-docker based env on your Windows system (for example, using docker desktop or multipass).

This is my env info:

```
> npm version
{
  npm: '8.19.4',
  node: '16.20.2',
  v8: '9.4.146.26-node.26',
  uv: '1.48.0',
  zlib: '1.2.12',
  brotli: '1.1.0',
  ares: '1.28.1',
  modules: '93',
  nghttp2: '1.61.0',
  napi: '8',
  llhttp: '6.0.11',
  openssl: '3.2.1',
  cldr: '44.1',
  icu: '74.2',
  tz: '2023c',
  unicode: '15.1'
}
```

**Comment by Pearl1594** [comment]
@SamEsmailR Is the package-lock.json file missing by any chance?

**Comment by lotus-biscoff29** [comment, marked as answer]
Hi, I am facing the same issue.

Background: `npm install` first failed with python3.12 as the default python interpreter. So I changed it to python3.11 and then it worked.
Then on doing `npm run serve' im getting the above error.

System details and env info: Ubuntu 24.04.2

$ npm --version
8.19.4


```
{
  npm: '8.19.4',
  node: '16.20.2',
  v8: '9.4.146.26-node.26',
  uv: '1.43.0',
  zlib: '1.2.11',
  brotli: '1.0.9',
  ares: '1.19.1',
  modules: '93',
  nghttp2: '1.47.0',
  napi: '8',
  llhttp: '6.0.11',
  openssl: '1.1.1v+quic',
  cldr: '41.0',
  icu: '71.1',
  tz: '2022f',
  unicode: '14.0',
  ngtcp2: '0.8.1',
  nghttp3: '0.7.0'
}

**Reply by lotus-biscoff29** [reply]
I'm able to setup cloudstack from scratch on Ubuntu 22.04.5 without any hiccup. Not sure why its breaking on Ubuntu 24.
